core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -236,12 +245,48 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
       leaderId = members.keys.headOption
   }
 
+  /**
+    * [For static members only]: Replace the old member id with the new one,
+    * keep everything else unchanged and return the updated member.
+    */
+  def maybeReplaceGroupInstance(oldMemberId: String,
+                                newMemberId: String,
+                                groupInstanceId: Option[String]): MemberMetadata = {
+    assert(groupInstanceId.isDefined)
```

Comment:
As we discussed before, better change `assert` to `throw IllegalStateException` with a meaningful error message; ditto below.

Reply:
Hmm.. does this function only have one caller who's already checked `group.hasStaticMember(groupInstanceId)`? In this case I think we can just name it `replaceGroupInstance` as it should always replace unless we have a bug.

Originally I was thinking there are multiple callers of it, and some may really turn into a no-op since it is not for static members, but now the call trace seems to indicate only one caller.